UI/cmake/os-windows.cmake

```diff
@@ -28,19 +30,21 @@ target_sources(
           update/update-window.hpp
           update/win-update.cpp
           update/win-update.hpp
-          update/models/branches.hpp
-          update/models/whatsnew.hpp
+          win-dll-blocklist.c
           win-update/updater/manifest.hpp)
 
 target_link_libraries(obs-studio PRIVATE crypt32 OBS::blake2 OBS::w32-pthreads MbedTLS::MbedTLS
                                          nlohmann_json::nlohmann_json Detours::Detours)
+
 target_compile_definitions(obs-studio PRIVATE PSAPI_VERSION=2)
-target_link_options(obs-studio PRIVATE /IGNORE:4098 /IGNORE:4099)
```

Comment:
Correct, instead of suppressing it globally I investigated the root causes and  selectively disabled the warning because it is only causes by a small subset of dependencies:

* `jansson` needs to be built with `JANSSON_STATIC_CRT`
  * Affected modules are `win-capture`, `rtmp-services`, `obs-outputs`, `ftl-sdk`, and `libobs`
* `zstd` seems to be hardcoded with `/DEFAULTLIB:LIBCMT`
  * Only affected module is `win-updater`
* `freetype` is hardcoded with `/DEFAULTLIB:MSVCRT`
  * Only affected module is `text-freetype2` 

So far it looks like only `jansson` can be compiled differently to fix the root cause, but once that is done we'd be left with `win-updater` and `text-freetype2` stuck with dependencies that use the "wrong" runtime libraries.

Reply:
Got it.

On the topic of Jansson, IIRC, setting `JANSSON_STATIC_CRT` ON causes it to use `/MT` and `/MTd`, which we found undesirable when we added Jansson to obs-deps. See https://github.com/obsproject/obs-deps/commit/0964f5e8eef57125b86d73799e2b5c7ea17d169a.